cumulusci/tasks/github/public_repo.py

```diff
@@ -0,0 +1,148 @@
+            "required": True,
+        },
+        "exclude": {"description": "A list of paths from repo root to exclude",},
+        "release_body": {
+            "description": "If True, the entire release body will be published to the public repo.  Defaults to False",
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.options["exclude"] = process_list_arg(self.options.get("exclude", []))
+        self.options["release_body"] = process_bool_arg(
+            self.options.get("release_body", False)
+        )
+        if self.options["version"] == "latest":
+            self.options["version"] = str(self.project_config.get_latest_version())
+        elif self.options["version"] == "latest_beta":
+            self.options["version"] = str(
+                self.project_config.get_latest_version(beta=True)
+            )
+
+    def _get_target_repo_api(self):
+        gh = self.project_config.get_github_api(
+            self._target_repo_info["owner"], self._target_repo_info["name"],
+        )
+        return gh.repository(
+            self._target_repo_info["owner"], self._target_repo_info["name"],
+        )
+
+    def _run_task(self):
+        self._target_repo_info = self.project_config._split_repo_url(
+            self.options["repo_url"]
+        )
+        self.target_repo = self._get_target_repo_api()
+
+        with temporary_dir(chdir=False) as target:
+            self._download_repo(target)
+            self._exclude_files(target)
+            commit = self._create_commit(target)
+            self._create_release(target, commit)
+
+    def _download_repo(self, path):
+        zf = download_extract_github_from_repo(
+            self.get_repo(),
+            ref="tags/{}".format(
+                self.project_config.get_tag_for_version(self.options["version"])
+            ),
+        )
+        with cd(path):
+            zf.extractall()
+
+    def _exclude_files(self, path):
+        with cd(path):
+            for path in self.options["exclude"]:
+                if os.path.isdir(path):
+                    shutil.rmtree(path)
+                elif os.path.isfile(path):
+                    os.remove(path)
+
+    def _create_commit(self, path):
+        committer = CommitDir(self.target_repo)
+        message = f"Publishing release {self.options['version']}"
+        commit = committer(path, self.options["branch"], commit_message=message)
+        return commit.sha
+
+    def _create_release(self, path, commit):
+        # Get current release info
+        tag_name = self.project_config.get_tag_for_version(self.options["version"])
+        repo = self.get_repo()
+        # Get the ref
+        try:
+            ref = repo.ref(f"tags/{tag_name}")
+        except github3.exceptions.NotFoundError:
+            message = f"Ref not found for tag {tag_name}"
+            self.logger.error(message)
+            raise GithubException(message)
```

Comment:
Why are you both logging at the error level and also raising an exception with the same message? Exceptions _should_ make it into the logs.